packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -1995,3 +2021,19 @@ fn new_js_string<'s>(
     let s = s.as_ref();
     v8::String::new(scope, s).expect(&format!("Could not create JS String: {s}"))
 }
+
+// Returns the new max heap size.
+extern "C" fn near_heap_limit_callback(
+    // This pointer is null, don't do anything with it.
+    _data: *mut std::ffi::c_void,
+    current_heap_limit: usize,
+    _initial_heap_limit: usize,
+) -> usize {
+    tracing::warn!(
+        "A JavaScript runner almost reached its heap limit! Use the '--js-runner-max-heap-size' \
```

Comment:
Can we be a little clearer with this message that it's a CLI argument passed when _running_ the engine?

Reply:
Clarified in https://github.com/hashintel/hash/pull/446/commits/289de04cd77ac79bac2ad06691b0c1a7feae19dc